extensions/amp-instagram/1.0/README.md

```diff
@@ -0,0 +1,148 @@
+
+### Web Component
+
+You must include each Bento component's required CSS library before adding custom styles in order to guarantee proper loading. Or use the lightweight pre-uprgrade styles available inline. See [Layout and Style](#layout-and-style).
+
+The examples below demonstrate use of the `<bento-instagram>` web component.
+
+#### Example: Import via npm
+
+[example preview="top-frame" playground="false"]
+
+Install via npm:
+
+```sh
+npm install @ampproject/bento-instagram
+```
+
+```javascript
+import '@ampproject/bento-instagram';
+```
+
+[/example]
+
+#### Example: Include via `<script>`
+
+[example preview="top-frame" playground="false"]
+
+```html
+<head>
+  <script async custom-element="bento-instagram" src="https://cdn.ampproject.org/v0/bento-instagram-1.0.js"></script>
+  <style data-bento-boilerplate>
+    bento-instagram {
+      display: block;
+      overflow: hidden;
+      position: relative;
+    }
+  </style>
+</head>
+<bento-instagram
+  id="my-instagram"
+  data-shortcode="CKXYAzuj7TE"
+  data-captioned>
+</bento-instagram>
+<button id="change-shortcode">Change shortcode</button>
+</div>
+
+<script>
+  (async () => {
+    const instagram = document.querySelector('#my-instagram');
+    await customElements.whenDefined('bento-instagram');
+
+    // set up button actions
+    document.querySelector('#change-shortcode').onclick = () => instagram.dataset.shortcode = '1totVhIFXl';
+  })();
+</script>
+```
+
+[/example]
+
+#### Layout and style
+
+Each Bento component has a small CSS library you must include to guarantee proper loading without [content shifts](https://web.dev/cls/). Because of order-based specificity, you must manually ensure that stylesheets are included before any custom styles.
+
+```html
+<link rel="stylesheet" type="text/css" href="https://cdn.ampproject.org/v0/bento-instagram-1.0.css">
+```
+
+Alternatively, you may also make the light-weight pre-upgrade styles available inline:
+
+```html
+<style data-bento-boilerplate>
+  bento-instagram {
+    display: block;
+    overflow: hidden;
+    position: relative;
+  }
+</style>
+```
+
+**Container type**
+
+The `bento-instagram` component has a defined layout size type. To ensure the component renders correctly, be sure to apply a size to the component and its immediate children (slides) via a desired CSS layout (such as one defined with `height`, `width`, `aspect-ratio`, or other such properties):
+
+```css
+bento-instagram {
+  height: 500px;
+  width: 100px;
+}
+```
+
+#### Attributes
+
+##### `data-shortcode`
+
+The instagram data-shortcode is found in every instagram photo URL. For example, in https://instagram.com/p/fBwFP, <code>fBwFP</code> is the data-shortcode.
+
+##### `data-captioned`
+
+Include the Instagram caption. `bento-instagram` will attempt to resize to the correct height including the caption.
+
+#### Styling
+
+You may use the `bento-instagram` element selector to style the component.
+
+### Preact/React Component
```

Comment:
This section is missing the `#### Layout and style` regarding sizing via `style` or `class` props.